### DISCUSSION TITLE
[2024-12-09] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#11812] [Weekly Demo Day]
* #11595
  The traces may be downloaded from summary page and viewed with `npx playwright show-trace
  ![image](https://github.com/user-attachments/assets/811927b7-7ce0-4055-96bd-0f46bc7a4237)
* Design updates
  * #11666
     1. Default values are italic
    ![image](https://github.com/user-attachments/assets/509a9d32-4263-4563-aad9-e7a9ee7bc039)
     2. The Text widget actually displays default values as placeholders (before it was a normal text).
    ![image](https://github.com/user-attachments/assets/73c32317-ac15-4f1d-8824-7cc445ae47ce)
     3. No more "grab" cursor over node's icon.
    ![image](https://github.com/user-attachments/assets/49939283-891e-4fc4-93bc-7d0c4ed7b295)
  * #11746
     ![image](https://github.com/user-attachments/assets/74da5807-7bac-4c18-a86a-c068cc19632f)
  * #11800
    ![image](https://github.com/user-attachments/assets/86b6fc6a-6ecc-41f1-9351-18c1b511776c)
* Bugfixes
  * #11684 - fixed an awful problem with visualization displaying wrong value
  * #11719 
  * #11744
  * #11739 - after waking from hibernation, GUI was creating two identical execution contexts.
  * #11761 - fixes an issue with node position being sometimes lost
  * #11797 - fixes compatibility issue between 2024.5 and 2024.4

**Comment by farmaazon** [comment]
@vitvakatu 
* #11582

  https://github.com/user-attachments/assets/fba44816-eed9-471d-83a7-8fe6e5892477

* #11715

  https://github.com/user-attachments/assets/4c6d7837-362e-4309-aa9b-5c0679d600e9

* #11695

  https://github.com/user-attachments/assets/a1233483-ee4a-47e4-84a1-64dd0b1505ef

  Roundtrip with Google Spreadsheets (shows non-trivial TSV data that includes quotes and newlines):

  https://github.com/user-attachments/assets/4ac662a2-809f-423a-9e47-628f46f92835

**Comment by kazcw** [comment]
GUI:

- #11690:

https://github.com/user-attachments/assets/14243102-3848-43fc-82bb-a48648536985

- #11753 (includes #11740):

https://github.com/user-attachments/assets/ff0eb399-73ea-4d68-a2ab-57d099a1dad5

- #11773:

https://github.com/user-attachments/assets/e87ba419-7465-48a7-bb08-101a76012a72

- #11803:

<img width="947" alt="Captura de ecrã 2024-12-09, às 07 16 53" src="https://github.com/user-attachments/assets/9165b6a7-f444-4837-b463-66230902ae5d">

- #11637 
- #11656 
- #11755
  - Fix #11745, and similar case for table editor widget
  - Enable AG Grid context menus on OS X
- #11768
- #11769
- #11786
- #11789
- #11791
- #11790

Enso language:
- #11671, #11682
  - Allows e.g. `3.14 : Integer : Text`
- #11754
  - Refactored parser to prevent similar bugs

**Comment by Frizi** [comment]
- https://github.com/enso-org/enso/pull/11655 finished, after a long battle with graph editor blinking on rename due to synchronization issues.

https://github.com/user-attachments/assets/10569d69-5831-4ad8-9d2c-2fd6f2b48182

- Improved the looks and consistency of our ag-grid widget and visualization. https://github.com/enso-org/enso/pull/11771

<img width="1139" alt="image" src="https://github.com/user-attachments/assets/7e69f211-22c5-42fd-8f6e-5e2017aaa15a">

- Improved edge coloring, fixed minor issues with edge positioning. https://github.com/enso-org/enso/pull/11810
<img width="238" alt="image" src="https://github.com/user-attachments/assets/e253b188-a6ce-468f-864c-9e814a8a2584">
<img width="140" alt="image" src="https://github.com/user-attachments/assets/d75ac860-c614-4f40-9cbc-b955e94f5a58">

- Attempted to fix a font issue causing https://github.com/enso-org/enso/issues/10692, gave up after a day of trying to wrap my head around the "Glyphs" app. Maybe we could use existing font instead?

**Comment by hubertp** [comment]
Relevant backend changes:
- Compute correct changeset when swapping nodes https://github.com/enso-org/enso/pull/11765
- Project Manager fails to create directory when one exists https://github.com/enso-org/enso/pull/11804
- Run workflows depending on changed files https://github.com/enso-org/enso/pull/11723
- Simplify shutdown logic on client disconnect in project-manager https://github.com/enso-org/enso/pull/11712
